### PR TITLE
feat(avaxc): avaxc sdk

### PR DESCRIPTION
**Pull request by aitarf54719747**
----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-268 Support Avax (C)](https://app.clickup.com/t/25577573/CW-268)

File: __mocks__/setup.js [hunk 1]
```diff
@@ -1,2 +0,0 @@
-console.debug = jest.fn(); // silence console
-console.log = jest.fn();
```
Comment (hank121314) [inline __mocks__/setup.js:2]: Unrelated changes.

File: packages/coin-avaxc/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      }
+    ]
+  },
+  externals: {
+    '@coolwallet/core': '@coolwallet/core',
+    rlp: 'rlp',
+    elliptic: 'elliptic',
+    'eth-sig-util': 'eth-sig-util'
+  }
+};
```
Comment (hank121314) [inline packages/coin-avaxc/webpack.config.js:35]: Unused file.

File: packages/coin-avaxc/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+export interface ERC20Transaction {
+  nonce: string,
+  gasPrice: string,
+  gasLimit: string,
+  to: string,
+  data: string,
+  value: string,
+	symbol: string,
+	decimals: string,
+	tokenSignature: string,
```
Comment (hank121314) [inline packages/coin-avaxc/src/utils.ts:22]:
```suggestion
  option: {
    info: {
      decimals: string,
      symbol: string
    }
  }
```

File: packages/coin-avaxc/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+      Buffer.from(r, 'hex'),
+      Buffer.from(s, 'hex')
+    );
+    const serializedTx = Buffer.from(rlp.encode(signedTransaction));
+
+    return `0x${serializedTx.toString('hex')}`;
+  };
+
+
+  signERC20Transaction = async (
```
Comment (hank121314) [inline packages/coin-avaxc/src/index.ts:155]: Should automatically apply official signature when data is erc20 transaction and `to` address is a contract address.
FYR
https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/911ca48ed646d9c63b7cb0cc96ea1ca1af39a4dc/packages/coin-matic/src/index.ts#L132-L148

File: packages/coin-avaxc/package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    "build": "npm run build:types && npm run build:ts",
+    "test": "jest --verbose"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.7.0",
+    "@babel/core": "^7.7.0",
+    "@babel/plugin-proposal-class-properties": "^7.7.4",
+    "@babel/preset-env": "^7.7.1",
+    "@babel/preset-typescript": "^7.7.7",
+    "@coolwallet/core": "file:../core",
```
Comment (hank121314) [inline packages/coin-avaxc/package.json:20]:
```suggestion
    "@coolwallet/core": "^1.1.5",
```

Change the location to npm registry.

File: packages/coin-avaxc/package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "dependencies": {
+    "@types/eth-sig-util": "^2.1.1",
+    "@types/keccak": "^3.0.1",
+    "ajv": "^8.10.0",
+    "bip32": "^2.0.4",
+    "eth-sig-util": "^3.0.1",
+    "keccak": "^3.0.2",
+    "rlp": "^2.2.7"
+  }
```
Comment (hank121314) [inline packages/coin-avaxc/package.json:39]:
```suggestion
  },
  "peerDependencies": {
     "@coolwallet/core": "^1.1.5",
  }
```

Should add `@coolwallet/core` as peerDependencies.

File: packages/coin-avaxc/src/package.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  "dependencies": {
+    "@types/eth-sig-util": "^2.1.1",
+    "@types/keccak": "^3.0.1",
+    "ajv": "^8.10.0",
+    "bip32": "^2.0.4",
+    "eth-sig-util": "^3.0.1",
+    "keccak": "^3.0.2",
+    "rlp": "^2.2.7"
+  }
+}
```
Comment (hank121314) [inline packages/coin-avaxc/src/package.json:40]: Should not have more than one `package.json` in a single npm repository.

**Review by hank121314 — CHANGES_REQUESTED** [review]
Please use prettier to fix the indentation and some empty lines.
And also provide some test cases in [coin-tester](https://github.com/CoolBitX-Technology/coolwallet-sdk/tree/master/packages/coin-tester).

File: packages/coin-tester/src/components/coins/avaxc/index.tsx [hunk 8]
```diff
@@ -0,0 +1,447 @@
...
+
+			const appId = localStorage.getItem('appId');
+			if (!appId) throw new Error('No Appid stored, please register!');
+			const signedTx = await temp.signTransaction(transport!, appPrivateKey, appId, 0, transaction);
+			await web3.eth.sendSignedTransaction(signedTx);
+			return signedTx;
+		}, setSignedTransaction);
+	};
+
+	const signERC20Transaction = async () => {
```
Comment (hank121314) [inline packages/coin-tester/src/components/coins/avaxc/index.tsx:77]: You can generate data without `abi`. Please refer to the other EVM compatible chain in `coin-tester`.

File: packages/coin-avaxc/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,507 @@
...
+
+		return `0x${serializedTx.toString('hex')}`;
+	};
+
+	signERC20Transaction = async (
+		transport: Transport,
+		appPrivateKey: string,
+		appId: string,
+		addressIndex: number,
+		transaction: Transaction
```
Comment (hank121314) [inline packages/coin-avaxc/src/index.ts:137]: Users should be able to pass their own symbol and decimals.
Please refer to the other EVM compatible chain in SDK.

**Review by hank121314 — CHANGES_REQUESTED** [review]
Please use prettier to fix the indentation.
Our indentation is using two spaces, not a tab.

**Review by hank121314 — CHANGES_REQUESTED** [review]
Please use prettier to fix the indentation.
Apply it to all changed files.
And remove some unrelated changes, please.
You can use https://github.com/CoolBitX-Technology/coolwallet-sdk/pull/326/files to find invalid indentations.
Also, the conflicts should be resolved.

File: packages/coin-tester/src/components/coins/avaxc/index.tsx [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+      };
+
+      const signedTx = await temp.signTypedData(transport!, appPrivateKey, appId, 0, typedData);
+
+      const recover = sigUtil.recoverTypedSignature_v4({
+        data: typedData,
+        sig: signedTx,
+      });
+
+      console.log('recover', recover);
```
Comment (hank121314) [inline packages/coin-tester/src/components/coins/avaxc/index.tsx:191]: Is there any need to add these lines?

**Review by hank121314 — CHANGES_REQUESTED** [review]

File: packages/coin-tester/src/components/coins/avaxc/index.tsx (hunk 10 above)
Comment (aitarf54719747) [reply packages/coin-tester/src/components/coins/avaxc/index.tsx:191]: This section is verifying that if signedTx is correct, recover will be equal to the wallet address

**Review by hank121314 — APPROVED** [review]

**Review by hankliu418 — APPROVED** [review]
